builder/common/access_config.go

```diff
@@ -281,6 +288,18 @@ func (c *AccessConfig) Session() (*session.Session, error) {
 		DecodeAuthZMessages(c.session)
 	}
 
+	// Append additional User-Agent products to the AWS SDK Go client.
+	userAgentProducts := []*awsbase.UserAgentProduct{
+		{Name: "APN", Version: "1.0"},
+		{Name: "HashiCorp", Version: "1.0"},
+		{Name: "Packer", Version: c.packerConfig.PackerCoreVersion, Extra: []string{"+https://www.packer.io"}},
+		{Name: "packer-plugin-amazon", Version: pluginversion.Version, Extra: []string{"+https://www.packer.io/docs/builders/amazon"}},
+	}
+	for i := len(userAgentProducts) - 1; i >= 0; i-- {
+		product := userAgentProducts[i]
+		c.session.Handlers.Build.PushFront(request.MakeAddToUserAgentHandler(product.Name, product.Version, product.Extra...))
+	}
```

Comment:
If this entire function relied on `awsbase` instead of reimplementing [`GetSession`](https://github.com/hashicorp/aws-sdk-go-base/blob/da4108d88728fcf4f5f9df9ea817285bccf7078e/session.go#L74) (like [this](https://github.com/hashicorp/terraform-provider-aws/blob/bc48c2178fff1c5b1b142481069e45fcb5870bd5/aws/config.go#L386)), I think it could be much cleaner.